Purchases/Public/RCPurchases.m

```diff
@@ -474,8 +474,8 @@ - (void)  logIn:(NSString *)appUserID
     [self.identityManager logInWithAppUserID:appUserID completion:^(RCPurchaserInfo *purchaserInfo,
                                                                     BOOL created,
                                                                     NSError * _Nullable error) {
-        CALL_IF_SET_ON_MAIN_THREAD(completion, purchaserInfo, created, error);
-
+        [self.operationDispatcher dispatchOnMainThread:^{ completion(purchaserInfo, created, error); }];
```

Comment:
I checked these one by one to make sure that we're not checking nullability in the ones that are not nullable